core/packet.c

```diff
@@ -109,31 +110,46 @@ static coap_status_t handle_request(lwm2m_context_t * contextP,
     coap_status_t result = NOT_FOUND_4_04;
 
     uriP = lwm2m_decode_uri(message->uri_path);
-    if (uriP == NULL) return BAD_REQUEST_4_00;
-
-    switch(uriP->flag & LWM2M_URI_MASK_TYPE)
-    {
+    if (uriP == NULL) {
```

Comment:
I meant in this special case. You need to check if the delete is on "/". Currently you do a handle_delete_all() for any invalid URI.
You need a special handling of "/" URI in lwm2m_decode_uri() setting a fleg that you'll test before branching below. The case uriP==NULL will return a Bad Request code.